### DISCUSSION TITLE
Best ways to patching machines hosting RKE2

### DISCUSSION THREAD

**Discussion by frit0-rb** [rancher/rke2#5126] [General]
Good afternoon,

I'm relatable new in Kubernetes and I wondering what is the best way to patch Linux (RHEL in this case).

Now I follow this steps to do it:
- First workers, one by one, draining and stopping rke2 services.
- Then controllers, one by one too, and draining them.

I don't know if there are best practices about this

**Comment by Martin-Weiss** [comment]
The OS patchlevel should not make any difference from the Kubernetes point of view. So it does not matter if you patch the OS on the controllers or workers, first.
Keep in mind that in a cloud-native world we really do not want to patch nodes - instead we want to replace them with a new OS template (that includes a newer patchlevel) - reason: upgraded /patched nodes are never 100% identical to new installed nodes.

P.S. If we can not re-deploy the nodes during upgrade - I personally patch the control plane first as this is the same for k8s updates.. but there is no real technical reason behind..
P.P.S: make sure to test the new patchlevel properly in staging before rolling it out to production ;-)

**Reply by frit0-rb** [reply]
Sorry, but i'm not talking about to update K8S.

My doubts coming to update only the OS. I'm not patching kubernetes at the same time as Linux.

That the point.

I there are any practices to make this procedure better to not affect kubernetes in major ways
